src/Compatibility/Microsoft.DotNet.PackageValidation/ApiCompatRunner.cs

```diff
@@ -31,63 +36,88 @@ public ApiCompatRunner(string noWarn, (string, string)[] ignoredDifferences, boo
         }
 
         /// <summary>
-        /// Runs the api compat for the tuples in the queue.
+        /// Runs the api compat for the tuples in the dictionary.
         /// </summary>
         public void RunApiCompat()
         {
-            foreach (var apicompatTuples in _queue.Distinct())
+            foreach (MetadataInformation left in _dict.Keys)
             {
-                // TODO: Add optimisations tuples.
-                using (Stream leftAssemblyStream = GetFileStreamFromPackage(apicompatTuples.leftAssemblyPackagePath, apicompatTuples.leftAssemblyRelativePath))
-                using (Stream rightAssemblyStream = GetFileStreamFromPackage(apicompatTuples.rightAssemblyPackagePath, apicompatTuples.rightAssemblyRelativePath))
+                Stream leftAssemblyStream = GetFileStreamFromPackage(_leftPackagePath, left.AssemblyId);
+                IAssemblySymbol leftSymbols = new AssemblySymbolLoader().LoadAssembly(left.AssemblyName, leftAssemblyStream);
+                leftAssemblyStream.Close();
```

Comment:
Maybe wrap the stream on a using statement for it to get disposed rather than closing it? That would help dispose it if there is an exception?